lib/Cro/WebApp/Template/Repository.rakumod

```diff
@@ -28,9 +28,14 @@ class Cro::WebApp::Template::Compiled is implementation-detail {
     has %.exports;
 
     #| Renders the template, setting the provided argument as the topic.
-    method render($topic --> Str) {
-        my $*TEMPLATE-REPOSITORY = $!repository;
-        &!renderer($topic)
+    method render($topic, :$fragment --> Str) {
+        if $fragment {
+            %.exports<fragment>{$fragment}($topic);
```

Comment:
Can you explain, why the `my $*TEMPLATE-REPOSITORY = $!repository;` isn't needed in this case?

Reply:
I am pretty sure that the compile phase loads the template `$.exports` on `Cro::WebApp::Template::Compiled.new` and that this attribute does not need adjusting according to the current`$!repository` whereas the `&.renderer` needs the repository path to pick the correct precompiled template.

That said, this statement is a little bit weak ... I can imagine that future evolutions of the code may expect both the exports and the repository attributes to have the same dependencies (especially in regard to global repository selectors) and that the current design is weak in this area. For example `$.exports` may one day be changed to a method, not a public attribute.

Therefore I have adjusted the code to put both arms of the if / else into the same global situation.